### PR TITLE
Include further keywords in `codemeta.json`

### LINKED ISSUE #4617: Improve codemeta.py script to include releveant keywords for codemeta.json
This issue is a followup to PR #4594 .

Here we should list the keywords we want to include on the codemeta.json file and include them in the scripts that updates the codemeta file produced by cffconvert.

### LINKED ISSUE #5095: Update `codemeta.py'
The code that generates the codemeta.json file should be updated in order to add:
  -  "programmingLanguage": "Python",
  -  "referencePublication": "https://doi.org/10.1051/0004-6361/202346488"

### PR DESCRIPTION
**Pull request by Astro-Kirsty**
This resolves #4617 and resolves #5095

Most of the keywords we want to add can be directly to the `json` file as they will not change across different versions. Do we actually need anything else defined in the `codemeta.py` if is the same across versions?
- Not sure about the `runtimePlatform`?
- In v1.0 we had: `dateCreated`, `datePublished` and `dateModified`. But now we do not use `dateModified`, how do we want to proceed?
- Do we want the `programmingLanguage` to be `Python` or `Python 3`

### CODE REVIEW

File: codemeta.json [hunk 1]
```diff
@@ -253,18 +253,31 @@
...
-        "Data analysis"
+        "Data analysis",
+        "Scientific/Engineering"
     ],
     "license": "https://spdx.org/licenses/BSD-3-Clause",
     "name": "Gammapy: Python toolbox for gamma-ray astronomy",
     "url": "https://gammapy.org/",
     "softwareVersion": "v1.2",
+    "applicationCategory": "Astronomy",
+    "applicationSubCategory": "Very-high energy gamma rays",
```
Comment (cgalelli) [inline codemeta.json:272]: Any possibility of being more broad here? e.g. HE gamma rays, VHE gamma rays, neutrino physics etc

**Review by registerrier — COMMENTED** [review]
Thanks @Astro-Kirsty . 
The json file is updated on the fly by the codemeta.py script. The idea is rather to adapt the script to include necessary changes, in particular the dates that have to be updated at the time of the release.

**Comment by Astro-Kirsty** [comment]
Okay @registerrier, but some of these are defined within the python file. For example, the key words. Is that because the file was edited afterwards and I should add anything that is missing? Or do I misunderstand something

**Review by bkhelifi — CHANGES_REQUESTED** [review]
Thanks a lot @Astro-Kirsty. I have some minor comments

**Review by bkhelifi — APPROVED** [review]
Thanks a lot!

**Review by registerrier — APPROVED** [review]
Thanks @Astro-Kirsty . Looks good.

I think the definition of publication date is a bit unclear in the codemeta specifications (since the parameter is associated to different versions). I would rather have e.g. the date of last LTS, which, I think, is more relevant than that of v0.1. Opinion @bkhelifi @adonath ?

**Comment by registerrier** [comment]
Merging now. We will see if further changes are needed in another PR.

**Comment by vuillaut** [comment]
> > Thanks @Astro-Kirsty . Looks good.
> > I think the definition of publication date is a bit unclear in the codemeta specifications (since the parameter is associated to different versions). I would rather have e.g. the date of last LTS, which, I think, is more relevant than that of v0.1. Opinion @bkhelifi @adonath ?
> 
> Not strongly! each release is a publication... But there are issues about that on which @vuillaut has worked: https://gitlab.in2p3.fr/escape2020/wp3/eossr/-/issues/13 and [codemeta/codemeta#285](https://github.com/codemeta/codemeta/pull/285). What do you recommand @vuillaut ? the date of the release (ie `time.now()`)?

Hi
I would recommend the date of the release but you could make a different choice that is more relevant to gammapy.

Concerning the keywords, I'd recommend you use the ones from https://astrothesaurus.org/ that has the advantage to be used across different services (but again, there is nothing *wrong* using a different vocabulary).

**Pull request merged**